forc-plugins/forc-doc/src/main.rs

```diff
@@ -77,8 +80,83 @@ pub fn main() -> Result<()> {
     }
     fs::create_dir_all(&doc_path)?;
 
-    // build core documentation
-    build_docs(&manifest, pkg_manifest, &doc_path, &build_instructions)?;
+    println!(
+        "   {} {} ({})",
+        "Compiling".bold().yellow(),
+        pkg_manifest.project_name(),
+        manifest.dir().to_string_lossy()
+    );
+
+    let member_manifests = manifest.member_manifests()?;
+    let lock_path = manifest.lock_path()?;
+    let plan = pkg::BuildPlan::from_lock_and_manifests(
+        &lock_path,
+        &member_manifests,
+        build_instructions.locked,
+        build_instructions.offline,
+    )?;
+
+    let type_engine = TypeEngine::default();
+    let decl_engine = DeclEngine::default();
+    let query_engine = QueryEngine::default();
+    let engines = Engines::new(&type_engine, &decl_engine, &query_engine);
+    let tests_enabled = true;
+    let mut compile_results = pkg::check(
+        &plan,
+        BuildTarget::default(),
+        build_instructions.silent,
+        tests_enabled,
+        engines,
+    )?;
+
+    if !build_instructions.no_deps {
+        let order = plan.compilation_order();
+        let graph = plan.graph();
+        let manifest_map = plan.manifest_map();
+        compile_results.reverse();
+
+        for node in order {
```

Comment:
Rather than reversing and popping from `compile_results` as we iterate, it looks like we should be able to iterate over the order and results together:

```suggestion
        for (node, compile_res) in order.iter().zip(compile_results) {
```